packages/at_sync_ui_flutter/lib/services/at_sync_ui_services.dart

```diff
@@ -138,6 +144,18 @@ class AtSyncUIService extends SyncProgressListener {
     );
   }
 
+  /// enables listening for sync progress
+  addSyncProgressListener() {
+    syncService?.addProgressListener(this);
+  }
+
+  /// stops listening for sync progress
+  ///
+  /// this should be called before switching atSign
+  removeSyncProgressListener() {
+    syncService?.removeProgressListener(this);
```

Comment:
@sachins-geekyants I am exploring options to do this at SDK layer, so app developers need not worry about clearing out listeners. I will keep you posted.

Reply:
Thanks @murali-shris.

Reply:
PR for the fix in SDK - https://github.com/atsign-foundation/at_client_sdk/pull/864
can you please test your scenario using switch_atsign_bug branch of at_client without the changes in this PR